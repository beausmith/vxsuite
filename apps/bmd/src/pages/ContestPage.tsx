import React, { useContext } from 'react'
import { RouteComponentProps } from 'react-router-dom'
import styled from 'styled-components'

import BallotContext from '../contexts/ballotContext'

import ButtonBar from '../components/ButtonBar'
import LinkButton from '../components/LinkButton'
import Main from '../components/Main'
import SingleCandidateContest from '../components/SingleCandidateContest'

interface ContestParams {
  id: string
}

interface Props extends RouteComponentProps<ContestParams> {}

const ContestPage = (props: Props) => {
  const { id } = props.match.params
  const { contests, updateVote, votes } = useContext(BallotContext)
  const currentContestIndex = contests.findIndex(x => x.id === id)
  const contest = contests[currentContestIndex]
  const prevContest = contests[currentContestIndex - 1]
  const nextContest = contests[currentContestIndex + 1]

  return (
    <React.Fragment>
      <Main>
        {contest ? (
          contest.type === 'plurality' && (
            <SingleCandidateContest
              contest={contest}
              vote={votes[contest.id]}
              updateVote={updateVote}
            />
          )
        ) : (
          <React.Fragment>
            <h1>Error</h1>
            <p>
              no contest exists for id <code>"{id}"</code>
            </p>
            <LinkButton to="/">Start Over</LinkButton>
          </React.Fragment>
        )}
      </Main>
      <ButtonBar>
        {nextContest ? (
          <LinkButton
            disabled={!nextContest}
            to={`/contests/${nextContest && nextContest.id}`}
          >
            Next
          </LinkButton>
        ) : (
          <LinkButton to="/summary">View Ballot</LinkButton>
        )}
        <LinkButton
          disabled={!prevContest}
          to={`/contests/${prevContest && prevContest.id}`}
        >
          Previous
        </LinkButton>
        <LinkButton to="/summary">View Summary</LinkButton>
      </ButtonBar>
    </React.Fragment>
  )
}

export default ContestPage
