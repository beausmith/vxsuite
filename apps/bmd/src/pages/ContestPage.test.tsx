import React from 'react'
import { Route } from 'react-router-dom'

import { render } from '../../test/testUtils'

import electionDefaults from '../data/electionDefaults.json'
import electionSample from '../data/electionSample.json'

const firstContestId = electionSample.contests[0].id

import ContestPage from './ContestPage'

it(`renders ContestPage`, () => {
  const { container } = render(
    <Route path="/contests/:id" component={ContestPage} />,
    {
      route: `/contests/${firstContestId}`,
    }
  )
  expect(container).toMatchSnapshot()
})

it(`renders error if no id match`, () => {
  const { container } = render(
    <Route path="/contests/:id" component={ContestPage} />,
    {
      route: '/contests/foobar',
    }
  )
  expect(container).toMatchSnapshot()
})

it(`displays accessible error if no id match`, () => {
  const { container } = render(
    <Route path="/contests/:id" component={ContestPage} />,
    {
      route: '/contests/foobar',
    }
  )
  expect(container).toMatchSnapshot()
})

it(`doesn't display help and settings pages if disabled`, () => {
  const { container, queryByText } = render(
    <Route path="/contests/:id" component={ContestPage} />,
    {
      election: {
        ...electionDefaults,
        ...electionSample,
        ...{
          bmdConfig: {
            showHelpPage: false,
            showSettingsPage: false,
          },
        },
      },
      route: `/contests/${firstContestId}`,
    }
  )
  expect(queryByText('Help')).toBeFalsy()
  expect(queryByText('Settings')).toBeFalsy()
  expect(container).toMatchSnapshot()
})
