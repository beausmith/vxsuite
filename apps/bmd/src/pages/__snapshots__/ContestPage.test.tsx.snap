// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays accessible error if no id match 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: scroll;
  padding: 1rem 0.5rem 2rem;
}

.c1 {
  width: 100%;
  max-width: 35rem;
  margin: 0 auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  padding: 0.5rem;
  background: #455a64;
  border-bottom: 1px solid darkgrey;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c3 > * {
  margin: 0.25rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c5 {
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  padding: 0.5rem;
  background: #455a64;
  border-bottom: 1px solid darkgrey;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c5 > * {
  margin: 0.25rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 {
  box-sizing: border-box;
  cursor: pointer;
  background: lightgrey;
  border: none;
  border-radius: 0.25rem;
  padding: 0.4rem 0.7rem;
  color: black;
  line-height: 1;
  white-space: nowrap;
}

.c4 {
  color: white;
  text-align: center;
}

@media (min-width:480px) {
  .c0 {
    padding: 2rem 1rem;
  }
}

@media print {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    padding: 0;
  }
}

@media print {
  .c1 {
    max-width: 100%;
    margin: 0;
  }
}

@media (min-width:480px) {
  .c3 > *:only-child {
    margin: auto;
    max-width: 30%;
  }
}

@media print {
  .c3 {
    display: none;
  }
}

@media (min-width:480px) {
  .c5 > *:first-child {
    margin-left: auto;
  }
}

@media (min-width:480px) {
  .c5 > * {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
  }
}

@media (min-width:480px) {
  .c5 > *:only-child {
    max-width: 30%;
  }
}

@media print {
  .c5 {
    display: none;
  }
}

@media print {
  .c4 {
    color: white;
  }
}

<div>
  <main
    class="c0"
  >
    <div
      class="c1"
    >
      <h1>
        Error
      </h1>
      <p>
        no contest exists for id 
        <code>
          "
          foobar
          "
        </code>
      </p>
      <button
        class="c2"
        type="button"
      >
        Start Over
      </button>
    </div>
  </main>
  <nav
    class="c3"
  >
    <button
      class="c2"
      type="button"
    >
      Next
    </button>
    <button
      class="c2"
      type="button"
    >
      Previous
    </button>
    <p
      class="c4"
    >
      0
       of 
      20
    </p>
  </nav>
  <nav
    class="c5"
  >
    <button
      class="c2"
      type="button"
    >
      Review
    </button>
    <button
      class="c2"
      type="button"
    >
      Help
    </button>
    <button
      class="c2"
      type="button"
    >
      Settings
    </button>
  </nav>
</div>
`;

exports[`doesn't display help and settings pages if disabled 1`] = `
.c3 {
  line-height: 1.2;
  max-width: 66ch;
}

.c3 h1 {
  line-height: 1.1;
  margin: 2rem 0 1rem;
}

.c3 h2 {
  margin: 1.5rem 0 1rem;
}

.c3 h3,
.c3 p {
  margin: 1rem 0;
}

.c3 h1 + h2 {
  margin-top: -0.75rem;
}

.c3 h1 + p,
.c3 h2 + p,
.c3 h3 + p {
  margin-top: -0.75rem;
}

.c3 > :first-child {
  margin-top: 0;
}

.c3 > :last-child {
  margin-bottom: 0;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  padding: 0.5rem;
  background: #455a64;
  border-bottom: 1px solid darkgrey;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c9 > * {
  margin: 0.25rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c12 {
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  padding: 0.5rem;
  background: #455a64;
  border-bottom: 1px solid darkgrey;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c12 > * {
  margin: 0.25rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c10 {
  box-sizing: border-box;
  cursor: pointer;
  background: lightgrey;
  border: none;
  border-radius: 0.25rem;
  padding: 0.4rem 0.7rem;
  color: black;
  line-height: 1;
  white-space: nowrap;
}

.c11 {
  color: white;
  text-align: center;
}

.c2 {
  font-weight: 600;
  text-transform: uppercase;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: scroll;
}

.c1 {
  width: 100%;
  max-width: 35rem;
  margin: 0px auto;
  padding: 1rem 0.75rem 0.5rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: scroll;
}

.c5:before {
  content: '';
  z-index: 1;
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
  height: 5px;
  width: 100%;
  position: fixed;
  opacity: 0;
  background: linear-gradient( to bottom, rgb(177,186,190) 0%, transparent 100% );
}

.c6 {
  display: grid;
  grid-auto-rows: minmax(auto,1fr);
  grid-gap: 0.75rem;
  width: 100%;
  max-width: 35rem;
  margin: 0 auto;
  padding: 0.5rem 0.5rem 2rem;
}

.c7 {
  cursor: pointer;
  position: relative;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.125rem;
  background: white;
  box-shadow: 0 0.125rem 0.125rem 0 rgba(0,0,0,0.14), 0 0.1875rem 0.0625rem -0.125rem rgba(0,0,0,0.12), 0 0.0625rem 0.3125rem 0 rgba(0,0,0,0.2);
  -webkit-transition: background 0.25s,color 0.25s;
  transition: background 0.25s,color 0.25s;
}

button.c7 {
  text-align: left;
}

.c7:focus-within {
  outline: -webkit-focus-ring-color auto 5px;
}

.c7:before {
  content: '';
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  background: white;
  border-right: 1px solid;
  border-color: lightgrey;
  width: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 2rem;
  border-radius: 0.125rem 0 0 0.125rem;
  color: #028099;
}

.c7 > div {
  word-break: break-word;
  padding: 0.5rem 0.5rem 0.5rem 4rem;
}

.c8 {
  margin-right: 0.5rem;
}

@media (min-width:480px) {
  .c3 {
    line-height: 1.3;
  }
}

@media (min-width:480px) {
  .c9 > *:only-child {
    margin: auto;
    max-width: 30%;
  }
}

@media print {
  .c9 {
    display: none;
  }
}

@media (min-width:480px) {
  .c12 > *:first-child {
    margin-left: auto;
  }
}

@media (min-width:480px) {
  .c12 > * {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
  }
}

@media (min-width:480px) {
  .c12 > *:only-child {
    max-width: 30%;
  }
}

@media print {
  .c12 {
    display: none;
  }
}

@media print {
  .c11 {
    color: white;
  }
}

@media (min-width:640px) {
  .c1 {
    padding: 1rem 1.5rem 0.5rem;
    padding-left: 5rem;
  }
}

@media (min-width:768px) {
  .c6 {
    padding-right: 1rem;
    padding-left: 1rem;
  }
}

@media (min-width:480px) {
  .c7 > div {
    padding: 1rem 1rem 1rem inherit;
  }
}

<div>
  <main
    class="c0"
  >
    <div
      class="c1"
    >
      <small
        class="c2"
      >
        Federal
        <span
          class="visually-hidden"
        >
          .
        </span>
      </small>
      <div
        class="c3"
      >
        <h1>
          President and Vice-President of the United States
          <span
            class="visually-hidden"
          >
            .
          </span>
        </h1>
        <p>
          <strong>
            Vote for 
            1
            .
          </strong>
           You have selected
           
          0
          .
        </p>
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <label
            class="c7"
            for="barchi-hallaren"
          >
            <input
              class="visually-hidden c8"
              id="barchi-hallaren"
              name="president"
              type="checkbox"
              value="barchi-hallaren"
            />
            <div
              class="c3"
            >
              <strong>
                Joseph Barchi and Joseph Hallaren
              </strong>
              <span
                class="visually-hidden"
              >
                ,
              </span>
              <br />
              Federalist
            </div>
          </label>
          <label
            class="c7"
            for="cramer-vuocolo"
          >
            <input
              class="visually-hidden c8"
              id="cramer-vuocolo"
              name="president"
              type="checkbox"
              value="cramer-vuocolo"
            />
            <div
              class="c3"
            >
              <strong>
                Adam Cramer and Greg Vuocolo
              </strong>
              <span
                class="visually-hidden"
              >
                ,
              </span>
              <br />
              People's
            </div>
          </label>
          <label
            class="c7"
            for="court-blumhardt"
          >
            <input
              class="visually-hidden c8"
              id="court-blumhardt"
              name="president"
              type="checkbox"
              value="court-blumhardt"
            />
            <div
              class="c3"
            >
              <strong>
                Daniel Court and Amy Blumhardt
              </strong>
              <span
                class="visually-hidden"
              >
                ,
              </span>
              <br />
              Liberty
            </div>
          </label>
          <label
            class="c7"
            for="boone-lian"
          >
            <input
              class="visually-hidden c8"
              id="boone-lian"
              name="president"
              type="checkbox"
              value="boone-lian"
            />
            <div
              class="c3"
            >
              <strong>
                Alvin Boone and James Lian
              </strong>
              <span
                class="visually-hidden"
              >
                ,
              </span>
              <br />
              Constitution
            </div>
          </label>
          <label
            class="c7"
            for="hildebrand-garritty"
          >
            <input
              class="visually-hidden c8"
              id="hildebrand-garritty"
              name="president"
              type="checkbox"
              value="hildebrand-garritty"
            />
            <div
              class="c3"
            >
              <strong>
                Ashley Hildebrand-MacDougall and James Garritty
              </strong>
              <span
                class="visually-hidden"
              >
                ,
              </span>
              <br />
              Whig
            </div>
          </label>
          <label
            class="c7"
            for="patterson-lariviere"
          >
            <input
              class="visually-hidden c8"
              id="patterson-lariviere"
              name="president"
              type="checkbox"
              value="patterson-lariviere"
            />
            <div
              class="c3"
            >
              <strong>
                Martin Patterson and Clay Lariviere
              </strong>
              <span
                class="visually-hidden"
              >
                ,
              </span>
              <br />
              Labor
            </div>
          </label>
        </div>
      </div>
    </div>
  </main>
  <nav
    class="c9"
  >
    <button
      class="c10"
      type="button"
    >
      Next
    </button>
    <button
      class="c10"
      type="button"
    >
      Previous
    </button>
    <p
      class="c11"
    >
      1
       of 
      20
    </p>
  </nav>
  <nav
    class="c12"
  >
    <button
      class="c10"
      type="button"
    >
      Review
    </button>
  </nav>
</div>
`;

exports[`renders ContestPage 1`] = `
.c3 {
  line-height: 1.2;
  max-width: 66ch;
}

.c3 h1 {
  line-height: 1.1;
  margin: 2rem 0 1rem;
}

.c3 h2 {
  margin: 1.5rem 0 1rem;
}

.c3 h3,
.c3 p {
  margin: 1rem 0;
}

.c3 h1 + h2 {
  margin-top: -0.75rem;
}

.c3 h1 + p,
.c3 h2 + p,
.c3 h3 + p {
  margin-top: -0.75rem;
}

.c3 > :first-child {
  margin-top: 0;
}

.c3 > :last-child {
  margin-bottom: 0;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  padding: 0.5rem;
  background: #455a64;
  border-bottom: 1px solid darkgrey;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c9 > * {
  margin: 0.25rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c12 {
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  padding: 0.5rem;
  background: #455a64;
  border-bottom: 1px solid darkgrey;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c12 > * {
  margin: 0.25rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c10 {
  box-sizing: border-box;
  cursor: pointer;
  background: lightgrey;
  border: none;
  border-radius: 0.25rem;
  padding: 0.4rem 0.7rem;
  color: black;
  line-height: 1;
  white-space: nowrap;
}

.c11 {
  color: white;
  text-align: center;
}

.c2 {
  font-weight: 600;
  text-transform: uppercase;
}

.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: scroll;
}

.c1 {
  width: 100%;
  max-width: 35rem;
  margin: 0px auto;
  padding: 1rem 0.75rem 0.5rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: scroll;
}

.c5:before {
  content: '';
  z-index: 1;
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
  height: 5px;
  width: 100%;
  position: fixed;
  opacity: 0;
  background: linear-gradient( to bottom, rgb(177,186,190) 0%, transparent 100% );
}

.c6 {
  display: grid;
  grid-auto-rows: minmax(auto,1fr);
  grid-gap: 0.75rem;
  width: 100%;
  max-width: 35rem;
  margin: 0 auto;
  padding: 0.5rem 0.5rem 2rem;
}

.c7 {
  cursor: pointer;
  position: relative;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.125rem;
  background: white;
  box-shadow: 0 0.125rem 0.125rem 0 rgba(0,0,0,0.14), 0 0.1875rem 0.0625rem -0.125rem rgba(0,0,0,0.12), 0 0.0625rem 0.3125rem 0 rgba(0,0,0,0.2);
  -webkit-transition: background 0.25s,color 0.25s;
  transition: background 0.25s,color 0.25s;
}

button.c7 {
  text-align: left;
}

.c7:focus-within {
  outline: -webkit-focus-ring-color auto 5px;
}

.c7:before {
  content: '';
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  background: white;
  border-right: 1px solid;
  border-color: lightgrey;
  width: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 2rem;
  border-radius: 0.125rem 0 0 0.125rem;
  color: #028099;
}

.c7 > div {
  word-break: break-word;
  padding: 0.5rem 0.5rem 0.5rem 4rem;
}

.c8 {
  margin-right: 0.5rem;
}

@media (min-width:480px) {
  .c3 {
    line-height: 1.3;
  }
}

@media (min-width:480px) {
  .c9 > *:only-child {
    margin: auto;
    max-width: 30%;
  }
}

@media print {
  .c9 {
    display: none;
  }
}

@media (min-width:480px) {
  .c12 > *:first-child {
    margin-left: auto;
  }
}

@media (min-width:480px) {
  .c12 > * {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
  }
}

@media (min-width:480px) {
  .c12 > *:only-child {
    max-width: 30%;
  }
}

@media print {
  .c12 {
    display: none;
  }
}

@media print {
  .c11 {
    color: white;
  }
}

@media (min-width:640px) {
  .c1 {
    padding: 1rem 1.5rem 0.5rem;
    padding-left: 5rem;
  }
}

@media (min-width:768px) {
  .c6 {
    padding-right: 1rem;
    padding-left: 1rem;
  }
}

@media (min-width:480px) {
  .c7 > div {
    padding: 1rem 1rem 1rem inherit;
  }
}

<div>
  <main
    class="c0"
  >
    <div
      class="c1"
    >
      <small
        class="c2"
      >
        Federal
        <span
          class="visually-hidden"
        >
          .
        </span>
      </small>
      <div
        class="c3"
      >
        <h1>
          President and Vice-President of the United States
          <span
            class="visually-hidden"
          >
            .
          </span>
        </h1>
        <p>
          <strong>
            Vote for 
            1
            .
          </strong>
           You have selected
           
          0
          .
        </p>
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <label
            class="c7"
            for="barchi-hallaren"
          >
            <input
              class="visually-hidden c8"
              id="barchi-hallaren"
              name="president"
              type="checkbox"
              value="barchi-hallaren"
            />
            <div
              class="c3"
            >
              <strong>
                Joseph Barchi and Joseph Hallaren
              </strong>
              <span
                class="visually-hidden"
              >
                ,
              </span>
              <br />
              Federalist
            </div>
          </label>
          <label
            class="c7"
            for="cramer-vuocolo"
          >
            <input
              class="visually-hidden c8"
              id="cramer-vuocolo"
              name="president"
              type="checkbox"
              value="cramer-vuocolo"
            />
            <div
              class="c3"
            >
              <strong>
                Adam Cramer and Greg Vuocolo
              </strong>
              <span
                class="visually-hidden"
              >
                ,
              </span>
              <br />
              People's
            </div>
          </label>
          <label
            class="c7"
            for="court-blumhardt"
          >
            <input
              class="visually-hidden c8"
              id="court-blumhardt"
              name="president"
              type="checkbox"
              value="court-blumhardt"
            />
            <div
              class="c3"
            >
              <strong>
                Daniel Court and Amy Blumhardt
              </strong>
              <span
                class="visually-hidden"
              >
                ,
              </span>
              <br />
              Liberty
            </div>
          </label>
          <label
            class="c7"
            for="boone-lian"
          >
            <input
              class="visually-hidden c8"
              id="boone-lian"
              name="president"
              type="checkbox"
              value="boone-lian"
            />
            <div
              class="c3"
            >
              <strong>
                Alvin Boone and James Lian
              </strong>
              <span
                class="visually-hidden"
              >
                ,
              </span>
              <br />
              Constitution
            </div>
          </label>
          <label
            class="c7"
            for="hildebrand-garritty"
          >
            <input
              class="visually-hidden c8"
              id="hildebrand-garritty"
              name="president"
              type="checkbox"
              value="hildebrand-garritty"
            />
            <div
              class="c3"
            >
              <strong>
                Ashley Hildebrand-MacDougall and James Garritty
              </strong>
              <span
                class="visually-hidden"
              >
                ,
              </span>
              <br />
              Whig
            </div>
          </label>
          <label
            class="c7"
            for="patterson-lariviere"
          >
            <input
              class="visually-hidden c8"
              id="patterson-lariviere"
              name="president"
              type="checkbox"
              value="patterson-lariviere"
            />
            <div
              class="c3"
            >
              <strong>
                Martin Patterson and Clay Lariviere
              </strong>
              <span
                class="visually-hidden"
              >
                ,
              </span>
              <br />
              Labor
            </div>
          </label>
        </div>
      </div>
    </div>
  </main>
  <nav
    class="c9"
  >
    <button
      class="c10"
      type="button"
    >
      Next
    </button>
    <button
      class="c10"
      type="button"
    >
      Previous
    </button>
    <p
      class="c11"
    >
      1
       of 
      20
    </p>
  </nav>
  <nav
    class="c12"
  >
    <button
      class="c10"
      type="button"
    >
      Review
    </button>
    <button
      class="c10"
      type="button"
    >
      Help
    </button>
    <button
      class="c10"
      type="button"
    >
      Settings
    </button>
  </nav>
</div>
`;

exports[`renders error if no id match 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: scroll;
  padding: 1rem 0.5rem 2rem;
}

.c1 {
  width: 100%;
  max-width: 35rem;
  margin: 0 auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  padding: 0.5rem;
  background: #455a64;
  border-bottom: 1px solid darkgrey;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c3 > * {
  margin: 0.25rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c5 {
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  padding: 0.5rem;
  background: #455a64;
  border-bottom: 1px solid darkgrey;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 > *:first-child {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c5 > * {
  margin: 0.25rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 {
  box-sizing: border-box;
  cursor: pointer;
  background: lightgrey;
  border: none;
  border-radius: 0.25rem;
  padding: 0.4rem 0.7rem;
  color: black;
  line-height: 1;
  white-space: nowrap;
}

.c4 {
  color: white;
  text-align: center;
}

@media (min-width:480px) {
  .c0 {
    padding: 2rem 1rem;
  }
}

@media print {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    padding: 0;
  }
}

@media print {
  .c1 {
    max-width: 100%;
    margin: 0;
  }
}

@media (min-width:480px) {
  .c3 > *:only-child {
    margin: auto;
    max-width: 30%;
  }
}

@media print {
  .c3 {
    display: none;
  }
}

@media (min-width:480px) {
  .c5 > *:first-child {
    margin-left: auto;
  }
}

@media (min-width:480px) {
  .c5 > * {
    -webkit-flex: 0;
    -ms-flex: 0;
    flex: 0;
  }
}

@media (min-width:480px) {
  .c5 > *:only-child {
    max-width: 30%;
  }
}

@media print {
  .c5 {
    display: none;
  }
}

@media print {
  .c4 {
    color: white;
  }
}

<div>
  <main
    class="c0"
  >
    <div
      class="c1"
    >
      <h1>
        Error
      </h1>
      <p>
        no contest exists for id 
        <code>
          "
          foobar
          "
        </code>
      </p>
      <button
        class="c2"
        type="button"
      >
        Start Over
      </button>
    </div>
  </main>
  <nav
    class="c3"
  >
    <button
      class="c2"
      type="button"
    >
      Next
    </button>
    <button
      class="c2"
      type="button"
    >
      Previous
    </button>
    <p
      class="c4"
    >
      0
       of 
      20
    </p>
  </nav>
  <nav
    class="c5"
  >
    <button
      class="c2"
      type="button"
    >
      Review
    </button>
    <button
      class="c2"
      type="button"
    >
      Help
    </button>
    <button
      class="c2"
      type="button"
    >
      Settings
    </button>
  </nav>
</div>
`;
